src/mender-update/daemon/context.cpp

```diff
@@ -473,45 +541,90 @@ static error::Error UnmarshalJsonStateData(const json::Json &json, StateData &st
 }
 
 expected::ExpectedBool Context::LoadDeploymentStateData(StateData &state_data) {
+	log::Trace("Loading the deployment state data");
+
 	auto &db = mender_context.GetMenderStoreDB();
 	auto err = db.WriteTransaction([this, &state_data](kv_db::Transaction &txn) {
 		auto exp_content = txn.Read(mender_context.state_data_key);
 		if (!exp_content) {
 			return exp_content.error().WithContext("Could not load state data");
 		}
-		auto &content = exp_content.value();
+		auto content = common::StringFromByteVector(exp_content.value());
 
-		auto exp_json = json::Load(common::StringFromByteVector(content));
+		auto exp_json = json::Load(content);
 		if (!exp_json) {
 			return exp_json.error().WithContext("Could not load state data");
 		}
 
+		log::Debug("Got database state data content: " + content);
+
 		auto err = UnmarshalJsonStateData(exp_json.value(), state_data);
 		if (err != error::NoError) {
-			if (err.code != make_error_condition(errc::not_supported)) {
-				return err.WithContext("Could not load state data");
-			}
+			return err.WithContext("Could not load state data");
+		}
+
+		switch (state_data.version) {
+		case 0:
+		case 1: {
+			log::Info("Got old state data version");
```

Comment:
```suggestion
			log::Info("Got state data version 1");
```
Maybe? "Old" sounds relative to today, in the future version 3 can also be old :)

Reply:
haha, as in we're `relatively old` now?